lib/ansible/modules/network/nxos/nxos_l2_interfaces.py

```diff
@@ -0,0 +1,279 @@
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for nxos_l2_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+  'metadata_version': '1.1',
+  'status': ['preview'],
+  'supported_by': 'network'
+}
+
+DOCUMENTATION = """
+---
+module: nxos_l2_interfaces
+version_added: 2.9
+short_description: Manages Layer-2 Interfaces attributes of NX-OS Interfaces
+description: This module manages Layer-2 interfaces attributes of NX-OS Interfaces.
+author: Trishna Guha (@trishnaguha)
+notes:
+  - Tested against NXOS 7.3.(0)D1(1) on VIRL
+options:
+  config:
+    description: A dictionary of Layer-2 interface options
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - Full name of interface, i.e. Ethernet1/1.
+        type: str
+        required: true
+      access:
+        description:
+          - Switchport mode access command to configure the interface as a
+            Layer-2 access.
+        type: dict
+        suboptions:
+          vlan:
+            description:
+            - Configure given VLAN in access port. It's used as the access
+              VLAN ID.
+            type: int
+      trunk:
+        description:
+          - Switchport mode trunk command to configure the interface as a 
+            Layer-2 trunk.
+        type: dict
+        suboptions:
+          native_vlan:
+            description:
+              - Native VLAN to be configured in trunk port. It is used as the
+                trunk native VLAN ID.
+            type: int
+          allowed_vlans:
+            description:
+              - List of allowed VLANs in a given trunk port. These are the only
+                VLANs that will be configured on the trunk.
+            type: str
+
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+    default: merged
+"""
+EXAMPLES = """
+# Using merged
+
+# Before state:
+# -------------
+#
+# interface Ethernet1/1
+#   switchport access vlan 20
+# interface Ethernet1/2
+#   switchport trunk native vlan 20
+# interface mgmt0
+#   ip address dhcp
+#   ipv6 address auto-config
+
+- name: Merge provided configuration with device configuration.
+  nxos_l2_interfaces:
+    config:
+      - name: Ethernet1/1
+        mode: trunk
+        trunk:  
+	  native_vlan: 10
+	  allowed_vlans: 2,4,15
```

Comment:
There's some mixed spaces and tabs here, please remove the tabs and use spaces only.